src/main/java/com/mykytam/courseapi/exceptions/CustomRestExceptionHandler.java

```diff
@@ -0,0 +1,55 @@
+package com.mykytam.courseapi.exceptions;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import java.time.LocalDateTime;
+
+@ControllerAdvice
+public class CustomRestExceptionHandler {
+
+    // Entity with certain id not found in DB
+    @ExceptionHandler(EntityNotFoundException.class)
+    public ResponseEntity<ApiError> entityNotFoundHandler(EntityNotFoundException ex) {
+        ApiError apiError = new ApiError(HttpStatus.NOT_FOUND, ex.getMessage(), LocalDateTime.now());
+        return new ResponseEntity<>(apiError, apiError.getStatus());
+    }
+
+    // Topic with certain id not found in DB
+    @ExceptionHandler(TopicNotFoundException.class)
+    public ResponseEntity<ApiError> topicNotFoundHandler(TopicNotFoundException ex) {
+        ApiError apiError = new ApiError(HttpStatus.NOT_FOUND, ex.getMessage(), LocalDateTime.now());
+        return new ResponseEntity<>(apiError, apiError.getStatus());
+    }
+
+    // Course with certain id not found in DB
+    @ExceptionHandler(CourseNotFoundException.class)
+    public ResponseEntity<ApiError> topicNotFoundHandler(CourseNotFoundException ex) {
+        ApiError apiError = new ApiError(HttpStatus.NOT_FOUND, ex.getMessage(), LocalDateTime.now());
+        return new ResponseEntity<>(apiError, apiError.getStatus());
+    }
+
+    // Student with certain id not found in DB
+    @ExceptionHandler(StudentNotFoundException.class)
+    public ResponseEntity<ApiError> topicNotFoundHandler(StudentNotFoundException ex) {
+        ApiError apiError = new ApiError(HttpStatus.NOT_FOUND, ex.getMessage(), LocalDateTime.now());
+        return new ResponseEntity<>(apiError, apiError.getStatus());
+    }
```

Comment:
You don't need these exception handlers since you have exception  handler for more generic `EntityNotFoundException`